### DISCUSSION TITLE
SCAM FLAG - False Dependabot email warning

### DISCUSSION THREAD

**Discussion by redevill** [community/community#63523] [Code Security]
### Select Topic Area

Bug

### Body

Just got a dependabot email warning me of a vulnerability in my library from "critters" package.
This email looks a LOT like a fishing scam - ESPECIALLY if it is NOT accurate.
Is it possible the new API is being MISUSED?

It claims that I am using "critters" and links to my package.json. With an "npm list critters" in my local and global repo, no such dependency exists.

I do have other Angular projects, where "build-angular" lib has a dependency on critters... but the one dependabot highlighted is incorrect.

TIA

**Comment by honeyankit** [comment]
- When Dependabot creates an security related PR from Dependabot alerts. The email sent via `Dependabot` looks like below. It contains the security vulnerability link which will directly take to `Dependabot alert` which resulted in creating this PR.
<img width="1130" alt="image" src="https://github.com/community/community/assets/5500812/4d990042-5457-4569-ba58-b2c839b16724">

After clicking on `security vulnerability link`
<img width="1289" alt="image" src="https://github.com/community/community/assets/5500812/42952dd5-4644-42f5-be84-70d2a0aced4d">

- Also in the end the email, there is link the PR, it commits messages and other details of the PR to verify. Clicking on these links takes me the correct file, commit messages in the PR. 
<img width="759" alt="image" src="https://github.com/community/community/assets/5500812/503ce0f7-bed1-438d-b552-b77b7b97801b">

**Reply by honeyankit** [reply]
> Just got a dependabot email warning me of a vulnerability in my library from "critters" package.

Could we check, what sender email name?

**Comment by redevill** [comment]
The Email I got from: <notifications@github.com>
![image](https://github.com/community/community/assets/8999255/ec30b45d-d105-4189-b1b7-ce7bda5c7e3b)

**Reply by redevill** [reply]
There was also a CC going to: security_alert@noreply.github.com

**Reply by redevill** [reply]
There was some content that was blocked by my email client...

**Reply by honeyankit** [reply]
I do see similar emails in my inbox as well. What do you get when you click on `View all alerts`? For few emails, when I click on the `View all alerts` page it takes me to the below:
<img width="1289" alt="image" src="https://github.com/community/community/assets/5500812/36cf316c-64ff-4b59-83e1-bca0355d5052">

**Reply by honeyankit** [reply]
For few, I get below message:
```bash
default backend - 404
```

**Reply by redevill** [reply]
I extracted the link from the email and pasted it in a separate window - to be sure... I shows me this page (except I modified username)
<img width="1338" alt="image" src="https://github.com/community/community/assets/8999255/26f3fc5c-3564-4f1a-bebb-b046b08d1c50">

**Comment by honeyankit** [comment]
My thoughts:
Since you are using "build-angular" in your other projects, which depends on `critters`, then `critters` becomes a transitive (or indirect) dependency of your project. When I see the security alert screenshot you provided, the security vulnerability is in `critters`.  

1. Ensure you run the below command in the directory of the project that Dependabot highlighted (I see you have already executed it)
```bash
npm list critters
```

2. Double-check the package-lock.json or yarn.lock in the flagged project. This file will give you the complete dependency tree, and you can see if "critters" is mentioned there and, if so, which package is introducing it.
package-lock.json

**Comment by redevill** [comment]
Think I found what dependabot was looking at.
The structure of the project in Github, has a building kind of 'host' and the distribution library that is put out to npm.
I only checked the published library - this does not have the critters dependency. 

However the build project that creates the distribution DOES have the dependency.
Thank you for the help / assurance!
